src/net/socket/udp_sock.c

```diff
@@ -74,12 +84,19 @@ static int udp_sendmsg(struct sock *sk, struct msghdr *msg, int flags) {
 
 	udp_build(skb->h.uh, sock_inet_get_src_port(sk), to->sin_port, total_len);
 
-	memcpy(skb->h.uh + 1, msg->msg_iov->iov_base, data_len);
+	size = sizeof(msg->msg_iov[i].iov_base[0]);
+	buf = buf_ = malloc(data_len);
```

Comment:
use of malloc in kernel is forbidden. In general, kmalloc is suited for this.
But in this case things are different. sk_buff widely used in network stack. It is buffer, optimized to hold network data. It have pointers to data on different layers, allows easy placing in queues, ...
Here, skb obtained via call to make_pack, which takes desired buffer length and returns length of actually allocated buffer. There is no need to allocate intermediate buffer and place data in it.

A loop is really required here, if actual_len is smaller than total_len. Then, make_pack/snd_pack should   be repeated until no data left